Wrappers/d3d8/IDirect3DDevice8.cpp

```diff
@@ -534,7 +534,7 @@ HRESULT m_IDirect3DDevice8::SetRenderState(D3DRENDERSTATETYPE State, DWORD Value
 		else // Main campaign
 		{
 			IsEnabledInRoom54 = false;
-			if (GetCutsceneID() == 0x4E || (GetSpecializedLight1() != 0x01 && GetSpecializedLight2() != 0x01))	// Exclude specialized lighting zone unless in specific cutscene
+			if (GetCutsceneID() == 0x4E || InSpecialLightZone)	// Exclude specialized lighting zone unless in specific cutscene
```

Comment:
I think this is suppose to exclude special light zones.

Shouldn't it be:
```c++
if (GetCutsceneID() == 0x4E || !InSpecialLightZone)
```

Reply:
Yep, silly typo. Now I'm not so convinced I was testing the right situations. We'll see what @Polymega says.

Reply:
Happy to help and test. I'll do this soon. With your new changes to the code, are there certain rooms/conditions in particular you can think for me to check, FrozenFish? Or just do a general sweep?

Reply:
Thanks @Polymega! I think a general sweep, plus a double check of the flickery transitions we dealt with back in [this PR](https://github.com/elishacloud/Silent-Hill-2-Enhancements/pull/181) should suffice.

Reply:
I haven't done a thorough test yet, but from quick testing, [this section of code](https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/master/Wrappers/d3d8/IDirect3DDevice8.cpp#L548-L552) doesn't appear to be working now. James' self shadow should be disabled until a certain camera cut happens, then it's restored for the remainder of the cutscene.

Reply:
Also, not a big deal, but if you want to be more accurate, I'd change the variable name from `IsEnabledInRoom54` to `IsEnabledForCutscene52` or something similar. (For this code, 0x54 is for a cutscene ID and not a room ID.)

Reply:
Ok, let's not worry about the duplicate SpecializedLight variables for now.  Just revert this back and I can reconcile these variables later, after the pull request is done.

Reply:
Reverted. Thanks for the assistance @Polymega.

Reply:
No problem. If the code changes get to a good point by this weekend and you guys are ready for me, I'll sit down and do a serious playtest at that time.

Reply:
@Polymega, below is the merged build with all of the changes from both branches.

Testing build:  [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6041083/d3d8.zip)

Reply:
So excited to give this a proper playthrough/testing soon! Thank you both very, very much. ❤️

Reply:
Took things nice and slow with my playthrough and everything looks great! Congratulations to you once more, FrozenFish! And thank you for the help and assistance, Elisha! 🥂

Reply:
Nice!  I also did some testing with `InSpecialLightZone` and `GetSpecializedLight1()` and `GetSpecializedLight2()` and they are all three referring to different things.  There does not seem to be any way to rationalize these.  So it looks like we are just going to have to keep different variables for each of these.

BTW: @Polymega, does this also fix the TV shadow issue?

Reply:
>BTW: @Polymega, does this also fix the TV shadow issue?

Yes, it does!

Reply:
Great to hear there's no new issues, thanks for all the help with this feature both of you!